integration-tests/stateful-pallet-storage/handleItemized.test.ts

```diff
@@ -0,0 +1,179 @@
+// Integration tests for pallets/stateful-pallet-storage/handleItemized.ts
+import "@frequency-chain/api-augment";
+import assert from "assert";
+import { createAndFundKeypair, generateDelegationPayload, signPayloadSr25519 } from "../scaffolding/helpers";
+import { KeyringPair } from "@polkadot/keyring/types";
+import { ExtrinsicHelper } from "../scaffolding/extrinsicHelpers";
+import { AVRO_CHAT_MESSAGE } from "../stateful-pallet-storage/fixtures/itemizedSchemaType";
+import { MessageSourceId, SchemaId } from "@frequency-chain/api-augment/interfaces";
+import { u16, u64 } from "@polkadot/types";
+
+describe("📗 Stateful Pallet Storage", () => {
+    let schemaId: SchemaId;
+    let schemaId_unsupported: SchemaId;
+    let msa_id: MessageSourceId;
+    let providerId: MessageSourceId;
+    let providerKeys: KeyringPair;
+
+    before(async function () {
+        // Create a provider for the MSA, the provider will be used to grant delegation
+        providerKeys = await createAndFundKeypair();
+        let createProviderMsaOp = ExtrinsicHelper.createMsa(providerKeys);
+        await createProviderMsaOp.fundAndSend();
+        let createProviderOp = ExtrinsicHelper.createProvider(providerKeys, "PrivateProvider");
+        let [providerEvent] = await createProviderOp.fundAndSend();
+        assert.notEqual(providerEvent, undefined, "setup should return a ProviderCreated event");
+        if (providerEvent && ExtrinsicHelper.api.events.msa.ProviderCreated.is(providerEvent)) {
+            providerId = providerEvent.data.providerId;
+        }
+        assert.notEqual(providerId, undefined, "setup should populate providerId");
+
+        // Create a schema for Itemized PayloadLocation
+        const createSchema = ExtrinsicHelper.createSchema(providerKeys, AVRO_CHAT_MESSAGE, "AvroBinary", "Itemized");
+        const [event] = await createSchema.fundAndSend();
+        if (event && createSchema.api.events.schemas.SchemaCreated.is(event)) {
+            [, schemaId] = event.data;
```

Comment:
This has been fixed now with #961 so you can do:
`schemaId = event.data.schemaId;`